
    <link rel="stylesheet" type="text/css" href="skin/pfl/css/prolinks.css" />

    <p>
      This colum presents the ProLinks method(s)
      responsible for inferring each functional linkage.
      See below for a brief description of each inference 
      method:
    </p>

      <table width="100%">
	<tr valign="top">
	  <th class="mp" align="right"> 
	    Rosetta Stone (RS)
	  </th>
	  <td class="mp" align="left"> 
	    uses gene fusion events in other organisms to 
            infer functional relatedness.
	  </td>
	</tr>
	<tr valign="top">
	  <th class="mp" align="right"> 
	    Phylogenetic<br/>Profiles (PP)
	  </th>
	  <td class="mp" align="left">
	    uses the pattern of presence and absence of 
            proteins across multiple genomes to detect 
	    functional linkages.
	  </td>
	</tr>
	<tr valign="top">
	  <th class="mp" align="right"> 
	    Gene Cluster (GC)
	  </th>
	  <td class="mp" align="left"> 
	    uses likely membership to operons in order 
            to infer functional linkages.
	  </td>
	</tr>
	<tr valign="top">
	  <th class="mp" align="right"> 
	    Gene Neighbor (GN)
	  </th>
	  <td class="mp" align="left"> 
	    uses both the chromosomal proximity of genes 
            and the conservation of such proximity across  
            multiple organisms to infer linkages.
	  </td>
	</tr>

	<tr valign="bottom">
	  <td class="fbfoot" align="left" valign="bottom">
	    <br /><br />
	    <a class="pl_general" 
	      href="http://genomebiology.com/2004/5/5/R35"
	      target="_blank"><i>Check the literature 
		for more details on each method...</i>
	    </a>
	  </td>
	</tr>
      </table>
